certora/specs/DoubleEndedQueue.spec

```diff
@@ -0,0 +1,240 @@
+import "helpers.spec"
+
+methods {
+    pushFront(bytes32)                    envfree
+    pushBack(bytes32)                     envfree
+    popFront()          returns (bytes32) envfree
+    popBack()           returns (bytes32) envfree
+    clear()                               envfree
+
+    // exposed for FV
+    begin()             returns (int128)  envfree
+    end()               returns (int128)  envfree
+    
+    // view
+    length()            returns (uint256) envfree
+    empty()             returns (bool)    envfree
+    front()             returns (bytes32) envfree
+    back()              returns (bytes32) envfree
+    at_(uint256)        returns (bytes32) envfree // at is a reserved word
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Helpers                                                                                                             │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+
+function absolute_diff(mathint a, mathint b) returns mathint {
+    return a > b ? a - b : b - a; 
+}
+
+function min_int128() returns mathint {
+    return max_uint256 / 2 + 1;
+}
+
+function max_int128() returns mathint {
+    return max_uint256 / 2;
+}
+
+// Could be broken in theory, but not in practice
+function beginLimited(int128 begin) returns bool {
+    return min_int128() < to_mathint(begin); // not inclusive so it can pushFront
```

Comment:
there is an issue here. `min_int128` is positive, and is bigger than the max value of `int128`. If you `require` that you are probably creating vacuity.

Reply:
I'm solving since this was completely changed, let me know if the vacuity is still there but I think we're now good